In [39]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [40]:
df_original = pd.read_csv('../data/ej3_tabla_p2.csv')
df = df_original.copy()

df

EDAD RIESGO EXAMEN
0    55   ALTO     SI
1    56  MEDIO     NO
2    58  MEDIO     SI
3    56   BAJO     NO
4    59   BAJO     NO
5    57  MEDIO     SI
6    60   BAJO     SI
7    53   ALTO     NO
8    59  MEDIO     SI

In [41]:
df['RIESGO'] = df['RIESGO'].map({'BAJO': 1, 'MEDIO': 2, 'ALTO': 3})
df['EXAMEN'] = df['EXAMEN'].map({'SI': 1, 'NO': 0})

In [42]:
print('Correlacion entre edad y riesgo:', df['EDAD'].corr(df['RIESGO']))


Correlacion entre edad y riesgo: -0.7150969419341943


Interpretación (r = −0.715):

- Es una correlación lineal negativa moderada–fuerte.

- Con tu codificación RIESGO: BAJO=1, MEDIO=2, ALTO=3, el signo “−” indica que a mayor EDAD, menor RIESGO (los más jóvenes tienden a tener riesgo más alto).

- La magnitud (~0.72) sugiere una relación importante: si ajustaras una recta, r² ≈ 0.51 ⇒ ~51% de la variación (numérica) del riesgo quedaría explicada por la edad bajo este modelo lineal.

In [43]:
from sklearn.preprocessing import MinMaxScaler
edad_scaler = MinMaxScaler().fit(df[['EDAD', 'RIESGO']])
df[['EDAD', 'RIESGO']] = edad_scaler.transform(df[['EDAD', 'RIESGO']])
X_train = df[['EDAD', 'RIESGO']].to_numpy()



In [44]:
# test_df = q = pd.DataFrame({"EDAD":[50, 51, 54, 55],
#                   "RIESGO":["BAJO","MEDIO","ALTO", "MEDIO"]})

# test_df['RIESGO'] = test_df['RIESGO'].map({'BAJO': 1, 'MEDIO': 2, 'ALTO': 3})
# test_df['EDAD'] = edad_scaler.transform(test_df[['EDAD']])

# X_test = test_df[['EDAD', 'RIESGO']].to_numpy()


# print("X_test shape:", X_test.shape)

# print("X_test:", X_test)







In [45]:
# #weights = np.array([0.0807, 0.074])
# weights = np.array([-0.076, -0.071])
# #bias = -0.0742
# bias = 0.071


# y_pred = X_test @ weights + bias
# score = (y_pred > 0).astype(int)

# print(y_pred)
# print(score)









In [46]:
## Prueba para cuestinario

weights = np.array([-0.076, -0.071])
bias = 0.071
y_pred = X_train @ weights + bias
score = (y_pred > 0).astype(int)

score

array([0, 1, 0, 1, 1, 0, 0, 0, 0])

In [47]:
df['EXAMEN']

0    1
1    0
2    1
3    0
4    0
5    1
6    1
7    0
8    1
Name: EXAMEN, dtype: int64